docs/examples/advanced_rules/accuracy_distribution_based.yaml

```diff
@@ -201,30 +196,9 @@ rule_bindings:
     entity_id: ACCURACY_CHECK_DISTRIBUTION_NOT_OK
     column_id: FULL_NAME
     row_filter_id: NONE
-    reference_columns_id: ACCURACY_CHECK_DISTRIBUTION_NOT_OK_REFERENCE_COLUMNS
```

Comment:
Why do we have all these "reference_columns" changes in a PR which is about supporting "failed records for custom SQL rule"? If not directly related, can we make the changes in separate PRs?

Reply:
1. I have removed reference_columns from rule_bindings, as this is optional so wanted to have few tests without specifying them.
2. Secondly, as per logic change we need to select the columns in custom_sql_statement that we are using in reference_columns. In that case we will have to update rule definition.

Reply:
Got it. Can we please separate the changes into two PRs then? One is for optional reference_columns. The other is for custom_sql_statement.

Reply:
If I include, reference columns then I have to update the rules, to select the columns used in "**reference_columns**" or select all columns in **custom_sql_statement**. So this change is for **custom_statement** itself which will also cover the optional **reference_columns** scenario.